engine/src/main/scala/cromwell/engine/db/ExecutionInfosByExecution.scala

```diff
@@ -0,0 +1,107 @@
+package cromwell.engine.db
+
+import cromwell.engine._
+import cromwell.engine.backend._
+import cromwell.engine.db.slick.{Execution, ExecutionInfo}
+import wdl4s.values.WdlFile
+
+case class ExecutionInfosByExecution(execution: Execution, executionInfos: Seq[ExecutionInfo]) {
+
+  lazy val callLogs: Option[CallLogs] = {
+    import ExecutionInfosByExecution._
+
+    case class Accumulator(stdout: Option[WdlFile], stderr: Option[WdlFile], logs: Map[String, WdlFile])
+
+    val acc = executionInfos.foldLeft(Accumulator(None, None, Map.empty)) { (acc, executionInfo) =>
```

Comment:
I'd love you even *more* than I already do if you were folding over a function, but you can limit yourself to your current level of jeff love if you'd prefer